test/unit/test-lambda.js

```diff
@@ -198,21 +201,132 @@ describe('Lambda Functions', async() => {
+            // +1 because it will also reset power to its initial value
+            expect(generatedValues.initConfigurations.length).to.be(powerValues.length + 1);
+        });
+        it('should generate an alias for each `num` and `powerValue` when `onlyColdStarts` is set', async() => {
+
+            const generatedValues = await invokeForSuccess(handler, { lambdaARN: 'arnOK', num: 5, onlyColdStarts: true});
+
+            // +1 because it will also reset power to its initial value
+            expect(generatedValues.initConfigurations.length).to.be((powerValues.length * 5) + 1);
         });
 
+    });
+
+    describe('publisher', async() => {
+
+        const handler = require('../../lambda/publisher').handler;
+
+        const invalidEvents = [
+            { },
+            {lambdaARN: 'arnOK'},
+            {lambdaARN: 'arnOK', powerValues: {}},
+            {lambdaARN: 'arnOK',
+                powerValues: {
+                    initConfigurations: [{
+                        powerValue: 512,
+                        alias: 'RAM512',
+                    }],
+                },
+            },
+            {lambdaARN: 'arnOK',
+                powerValues: {
+                    iterator: {
+                        index: 1,
+                        count: 1,
+                    },
+                },
+            },
+            {lambdaARN: 'arnOK',
+                powerValues: {
+                    initConfigurations: [{
+                        powerValue: 512,
+                        alias: 'RAM512',
+                    }, {
+                        powerValue: 1024,
+                        alias: 'RAM1024',
+                    }],
+                    iterator: {
+                        index: 2,
+                        count: 3,
+                    },
+                },
+            },
+            {lambdaARN: 'arnOK',
+                powerValues: {
+                    initConfigurations: [{
+                        powerValue: 512,
+                        alias: 'RAM512',
+                    }, {
+                        powerValue: 1024,
+                        alias: 'RAM1024',
+                    }],
+                    iterator: {
+                        index: 3,
+                        count: 2,
+                    },
+                },
+            },
+        ];
+
+        invalidEvents.forEach(async(event) => {
+            it('should explode if invoked with invalid payload - ' + JSON.stringify(event), async() => {
+                await invokeForFailure(handler, event);
+            });
+        });
+
+        it('should publish the given lambda version', async() => {
+
+            const aliasValue = 'RAM512';
+            const originalIndex = 0;
+            const generatedValues = await invokeForSuccess(handler, {
+                lambdaARN: 'arnOK',
+                powerValues: {
+                    initConfigurations: [{
```

Comment:
This default event structure seems repeated quite a few times, it's probably worth to define as a variable and then reuse for multiple tests.